src/librustc_mir/transform/remove_dead_blocks.rs

```diff
@@ -72,11 +72,11 @@ fn retain_basic_blocks(mir: &mut Mir, keep: &BitVector) {
         if alive_index != used_blocks {
             // Swap the next alive block data with the current available slot. Since alive_index is
             // non-decreasing this is a valid operation.
-            mir.basic_blocks.swap(alive_index, used_blocks);
+            mir.cfg.basic_blocks.swap(alive_index, used_blocks);
```

Comment:
I prefer having `RemoveDeadBlocks` being a "GC" pass and doing the swapping "manually", with other passes not removing BBs.